src/Umbraco.Core/Routing/PublishedRouter.cs

```diff
@@ -714,22 +714,24 @@ private void FindTemplate(IPublishedRequestBuilder request, bool contentFoundByF
 
             // TODO: We need to limit altTemplate to only allow templates that are assigned to the current document type!
             // if the template isn't assigned to the document type we should log a warning and return 404
-            var templateId = request.PublishedContent.TemplateId;
-            ITemplate? template = GetTemplate(templateId);
-            request.SetTemplate(template);
-            if (template != null)
+            if (request.PublishedContent.TemplateId is int templateId && templateId != default)
```

Comment:
Hi @PerplexDaniel,

Is there a performance reason to do this over something like:

```suggestion
            if (request.PublishedContent.TemplateId.HasValue && request.PublishedContent.TemplateId != default(int))
```

Which to me feels more instinctive to me.

Not a criticism, curious to learn!

Reply:
Not really, I like the `<property> is <type> var` syntax to do a `null` check and simultaneously assign the non null value to a variable so you do not have to access the property again. But feel free to change to the other syntax if that's more clear.

Reply:
Good point. I see what you've done and it makes sense!